### PR TITLE
add fade between depths in depth reticle

### PR DESCRIPTION
**Pull request by ZappoMan**
instead of snapping the reticle instantly between different depths, fade over a short time period. also fix incorrect rotation in reticle when at depth.

### CODE REVIEW

**Comment by ZappoMan** [comment]
fixes this bug - https://app.asana.com/0/26320649018758/93278522205625

Testing notes:
1) run interface in hmd
2) run this script: https://rawgit.com/ZappoMan/hifi/depthReticleWork/examples/depthReticle.js
3) notice that when the mouse cursor/reticle moves over an overlay or entity that it fades to the new depth instead of snapping
4) stop this version of depthReticle.js and compare against the current one on S3, notice that it snaps.

This fade will help people adjust their vision particularly when there is a fine gap between two very large depths.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6d9c5856041ffd26eb1f06bbe03a463fb6fecb47 available at: https://gist.github.com/b345a754f32ca32769d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7212.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7212.dmg

**Comment by judasshuffle** [comment]
Tested, Yes it works. I had no idea what you meant till i tried it The cursor sorta inny outty stop me if I'm being too technical. its amazing how much of this stuff would never occur to me as wrong until I try something that makes more sense.It gives a much clearer idea of what you are about to  on
unrelated Its difficult to access the top bar menus to get out of rift mode

File: examples/depthReticle.js [hunk 1]
```diff
@@ -37,11 +44,39 @@ Script.update.connect(function(deltaTime) {
...
+        }
+    }
+
+    // move the reticle toward the desired depth
+    if (desiredDepth != Reticle.depth) {
+
+        // determine the time between now, and when we set our determined our desiredDepth
+        var elapsed = now - desiredDepthLastSet;
+        var distanceToFade = desiredDepth - wasDepth;
+        var percentElapsed = Math.min(1, elapsed / TIME_TO_FADE_DEPTH);
```
Comment (PhilipRosedale) [inline examples/depthReticle.js]: Please make this non-linear, you know how uncle philip feels about linear.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e403c990d96f8c174ecb97b82e0ed906b7ed5159 available at: https://gist.github.com/b345a754f32ca32769d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7212.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7212.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 188cf3694b83f36a6808bac3659ff7f1958f2f99 available at: https://gist.github.com/b345a754f32ca32769d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7212.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7212.dmg

**Pull request merged**